src/artm/regularizer/decorrelator_phi.cc

```diff
@@ -63,6 +90,33 @@ google::protobuf::RepeatedPtrField<std::string> DecorrelatorPhi::class_ids_to_re
   return config_.class_id();
 }
 
+void DecorrelatorPhi::UpdateTopicPairs(const DecorrelatorPhiConfig& config) {
+  config_.clear_first_topic_name();
+  config_.clear_second_topic_name();
+  config_.clear_value();
+
+  topic_pairs_.clear();
+  int topics_len = config.first_topic_name_size();
+  if (topics_len) {
+    if (topics_len != config.second_topic_name_size() ||
+      topics_len != config.value_size()) {
+      BOOST_THROW_EXCEPTION(::artm::core::CorruptedMessageException(
```

Comment:
No, this is not a fix, it'a a description of error. Also I don't want to move into descr section, because it is not a message, that can pass directly through c_interface.

Error checking in SmoothSparsePhi is the same as here.